app.js

```diff
@@ -0,0 +1,87 @@
+var express = require('express')
+var http = require('http')
+var path = require('path')
+var passport = require('passport')
+var LocalStrategy = require('passport-local').Strategy
+
+//==================================================================
+// Define the strategy to be used by PassportJS
+passport.use(new LocalStrategy(
+  function(username, password, done) {
+    // @todo Tie this authentication to the members database in CouchDB
+    if (username === "admin" && password === "admin") // stupid example
+      return done(null, {name: "admin"})
+
+    return done(null, false, { message: 'Incorrect username.' })
+  }
+))
+
+// Serialized and deserialized methods when got from session
+passport.serializeUser(function(user, done) {
+    done(null, user)
+})
+
+passport.deserializeUser(function(user, done) {
+    done(null, user)
+})
+
+//==================================================================
+// Start express application
+var app = express()
+
+// all environments
+app.set('port', process.env.PORT || 3000)
+app.use(express.favicon())
+app.use(express.logger('dev'))
+app.use(express.cookieParser()) 
+app.use(express.bodyParser())
+app.use(express.methodOverride())
+app.use(express.session({ secret: 'securedsession' }))
+app.use(passport.initialize()) // Add passport initialization
+app.use(passport.session())    // Add passport initialization
+app.use(app.router)
+app.use(express.static(path.join(__dirname, 'public')))
+
+// development only
+if ('development' == app.get('env')) {
+  app.use(express.errorHandler())
+}
+
+//==================================================================
+// api 
+
+// @todo this will only work for GET operations, need to also do DELETE,
+// PUT, and POST.
+// http://stackoverflow.com/questions/18728039/express-req-pipe-does-not-work
+app.get('/api/*', function(req, res) {
+  var endpoint = require('api/' + req.path[1])
+  endpoint.auth(req, res, function() {
+    var apiUrl = 'http://localhost:5984';
+    var url = apiUrl + req.url;
+    req.pipe(request(url).pipe(res));
+  })
+})
```

Comment:
This is where the magic of the how the auth functions in the api endpoint files is used for protecting CouchDB from unauthorized operations. Right now this only works for GET operations but we could write more code to handle PUT, POST, and DELETE operations.